tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +217,224 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
       assert.include hls, anchor12.highlight[anchor12.startPage]
       assert.include hls, anchor2.highlight[anchor2.startPage]
       assert.notInclude hls, anchor3.highlight[anchor3.startPage]
+
+  describe 'two-phased anchoring', ->
+
+    class DummyDocumentAccess
+
+      @applicable: -> true
+
+      isPageMapped: (index) -> index in @_rendered
+
+      # Helper function to trigger a page rendering
+      renderPage: (index) ->
+        return if @isPageMapped index
+
+        @_rendered.push index
+
+        # Publish an event
+        event = document.createEvent "UIEvents"
+        event.initUIEvent "docPageMapped", false, false, window, 0
+        event.pageIndex = index
+        window.dispatchEvent event
+
+      # Helper function to trigger a page unrendering
+      unrenderPage: (index) ->
+        return unless @isPageMapped index
+
+        i = @_rendered.indexOf index
+        @_rendered.splice(i, 1)
+
+        # Publish an event
+        event = document.createEvent "UIEvents"
+        event.initUIEvent "docPageUnmapped", false, false, window, 0
+        event.pageIndex = index
+        window.dispatchEvent event
```

Comment:
These helper functions are not part of the `DummyDocumentAccess` object so it would be clearer to just move them into helpers that accept a `document`. e.g.

```coffee
unrenderPage: (doc, index) ->
  return unless doc.isPageMapped(index)
  ...
```